test/receiver/receiver.go

```diff
@@ -65,13 +66,43 @@ func main() {
 	outDir := flag.String("outdir", "", "Output directory. Absolute path. Optional.")
 	profiler := flag.String("profiler", "", "Where should the profiler listen?")
 	exitAfter := flag.Duration("exitAfter", time.Second*10, "Exit after not receiving any message for this time. Will work only of outDir == ''. Will not exit if Duration is 0")
+	localAPIPortFlag := flag.Int64("local-api-port", 8024, "specify which port the local HTTP API should be listening on")
 
 	flag.Parse()
 
 	if *portsStr == "" {
 		log.Fatal("please supply the ports argument")
 	}
 
+	type status struct {
+		Ready        bool
+		IdleTimeSecs int64
+	}
+
+	var ready bool = false
+	var dataReceived bool = false
```

Comment:
Maybe just server timestamp since last data received and 0 otherwise?

Reply:
you still need a bolean as a different variable you can't use the same variable. But it should be renamed dataProcessed actually, not dataReceived. Because of all the buffering, data might be sent 100 secs after being received.